packages/ui/routes/mailings/__partials/mailings-selection-actions.vue

```diff
@@ -96,6 +143,56 @@ export default {
       }
       this.closeMoveManyMailsDialog();
     },
+    handleInitDownload({
+      withFtp,
+      mailingIds,
+      havePreview,
+      mailsWithoutPreviewSelection,
+    }) {
+      if (havePreview) {
+        this.handleDownloadMailSelections({ withFtp, mailingIds });
+        return;
+      }
+      this.$refs.downloadMailsDialog.open({
+        withFtp,
+        mailingIds,
+        mailsWithoutPreviewSelection,
+      });
+    },
+    handleInitSingleDownload({ withFtp, mailing }) {
+      const mailingIds = [mailing.id];
+      const havePreview = mailing.hasHtmlPreview;
+      this.handleInitDownload({ withFtp, mailingIds, havePreview });
+    },
+    handleInitMultipleDownload({ withFtp }) {
+      const filteredMailingsIds = this.mailingsSelection
+        .filter((el) => !this.mailsWithoutPreviewSelection.includes(el))
+        ?.map((mail) => mail?.id);
+      const allHavePreview = this.mailsWithoutPreviewSelection?.length <= 0;
```

Comment:
`isAllHavePreview`

Reply:
This is plural so we don't say `isAllHavePreview` `allHavePreview` works fine in our case

Reply:
https://betterprogramming.pub/useful-tips-for-naming-your-variables-8139cc8d44b5